src/swcio.hpp

```diff
@@ -30,384 +34,213 @@ class swc_record {
         custom
     };
 
+    kind type = kind::undefined; // record type
```

Comment:
Yes; `swc_record` now is a record-like struct with full r/w access by users of the struct. It can represent a record in a sequence of SWC lines from a file even if those records in the file are strictly speaking ill-formed.

Validation/consistency checking is performed separately.